app/Http/Controllers/Admin/CategoriesController.php

```diff
@@ -0,0 +1,118 @@
+    {
+        // update Meetings Status
+        $this->meeting->updateStatus();
+        
+        $category = $this->category->withTrashed()->findOrFail($id);
+        $meetings = $category->meetings()->withTrashed()->latest()->paginate(10);
+        return view('admin.chatrooms.categories.show')
+            ->with('category', $category)
+            ->with('meetings', $meetings)
+            ->with('statusColor', $this->meeting->statusColor());
+    }
+    public function add()
+    {
+        return view('admin.chatrooms.categories.add');
+    }
+    public function edit($id)
+    {
+        $category = $this->category->withTrashed()->findOrFail($id);
+        return view('admin.chatrooms.categories.edit')
+        ->with('category', $category);
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'name'          => 'required|min:1|max:20|unique:categories,name',
+            'color'         => 'required|regex:/^#[0-9a-fA-F]{3}([0-9a-fA-F]{3})?$/', //Hex Color
+            'icon'          => 'required|mimes:jpg,jpeg,png,gif',
+            'description'   => 'required|min:5|max:100'
+        ]);
+
+        $this->category->name = str_replace(" ", "_", ucwords($request->name));
+        $this->category->color = $request->color;
+        $this->category->description = $request->description;
+        $this->category->icon = $this->saveIcon($request);
+        $this->category->save();
+        return redirect()->route('admin.chatroom.category.index');
+    }
+    public function update(Request $request, $id)
+    {
+        $category = $this->category->withTrashed()->findOrFail($id);
+        $request->validate([
+            'name' => 'required|min:1|max:20|unique:categories,name,' . $category->id,
+            'color'=> 'required|regex:/^#[0-9a-fA-F]{3}([0-9a-fA-F]{3})?$/',
+            'icon' => 'mimes:jpg,jpeg,png,gif',
+            'description' => 'required|min:5|max:100'
+        ]);
+        
+        $category->name = str_replace(" ", "_", ucwords(strtolower($request->name)));
+        $category->color = $request->color;
+        $category->description = $request->description;
+        if ($request->icon) {
+            $this->deleteIcon($id);
+            $category->icon = $this->saveIcon($request);
+        }
+        $category->save();
+        return redirect()->route('admin.chatroom.category.index');
+    }
+    public function delete($id)
+    {
+        $category = $this->category->findOrFail($id);
+        foreach ($category->meetings as $meeting) {
+            $meeting->delete();
+        }
+        $category->delete();
+
+        return redirect()->route('admin.chatroom.category.index');
+    }
+    public function restore($id)
+    {
+        $category = $this->category->withTrashed()->findOrFail($id);
+        $category->restore();
+        return redirect()->route('admin.chatroom.category.index');
+    }
+
+    public function deleteIcon($id)
```

Comment:
maybe change the function name because it says `deleteIcon` but you are actually deleting the category. Your function should also be related to what it's doing inside.

Reply:
This function is called when updating a category image to delete old image.
In this code, specify a record and get icon name, then Try to delete the image.

Reply:
I suggest to make it more clear, like `deleteCategoryIcon` instead of just `deleteIcon`. Your function name does not relate to the category model.

Reply:
I changed fuction names
saveIcon -> saveCategoryIcon
deleteIcon -> deleteCategoryIcon